src/znc.cpp

```diff
@@ -628,6 +629,7 @@ bool CZNC::WriteNewConfig(const CString& sConfigFile) {
     VCString vsLines;
 
     vsLines.push_back(MakeConfigHeader());
+    vsLines.push_back("PidFile = /run/znc/znc.pid");
```

Comment:
I don't have a real say, but I'm going to -1 your entire pull req.  The default pidfile should be able to be configured *without* having to do `./configure` and then compiling to do it, if we want a customized PIDFile location, it would be prudent to put it into a config-level-defined option, rather than as a compile option (precedent: Apache, nginx, lighttpd, many other services which permit pidfile paths to be defined at the configuration file level, rather than at the configure command).

As psychon identified, this will break as-is if we want an installation in `~/znc-install`.  If the intent is to run this as a system service, then this will work for that; but as there is zero guarantee the system will have ZNC running as a system service; with many people using user-space installations, I don't see this being a 'compatible' change being introduced.

We may also have cases where people take ZIP files from one system, and move them to another with different user/folder paths; we have to make sure that won't break, especially if we need users to recompile each and every time...